src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -96,13 +128,90 @@ private void RootCommandHandler(FileInfo templateFilePath, FileInfo parametersFi
                 }
 
                 Console.WriteLine($"{IndentedNewLine}Rules passed: {passedEvaluations}");
+
+                return 1;
             }
             catch (Exception exp)
             {
                 Console.WriteLine($"An exception occured: {GetAllExceptionMessages(exp)}");
+                return -1;
             }
         }
 
+        private void AnalyzeDirectory(DirectoryInfo directoryPath)
+        {
+            try {
+
+                if (!directoryPath.Exists)
+                {
+                    Console.WriteLine($"Invalid directory ({directoryPath})");
+                    return;
+                }
+
+                // Find files to analyze
+                List<FileInfo> filesToAnalyze = new List<FileInfo>();
+                FindJsonFilesInDirectoryRecursive(directoryPath, filesToAnalyze);
+
+                // Log root directory
+                string directoryMetadata = Environment.NewLine + Environment.NewLine + $"Directory: {directoryPath}";
+                Console.WriteLine(directoryMetadata);
```

Comment:
minor: you can just create the string to write directly inside the function call instead of saving to a variable first.